test/e2e/external_gateways.go

```diff
@@ -448,7 +448,8 @@ var _ = ginkgo.Describe("External Gateway", func() {
 
 					ginkgo.By(fmt.Sprintf("Verifying connectivity to the pod [%s] from external gateways", addresses.srcPodIP))
 					for _, gwContainer := range gwContainers {
-						_, err := runCommand(containerRuntime, "exec", gwContainer, "ping", "-c", testTimeout, addresses.srcPodIP)
+						// Ping from a common IP address that exists on both gateways to ensure test coverage where ingress reply goes back to the same host.
```

Comment:
old traffic path was: externalGWContainer to pod inside the cluster ingress ping test ? and new traffic path check is 
```
ping test to use a common
IP address that is emulated on a loopback acting as though it lives
behind the gateways. This ensures that packets sent from either gateway
will have the same hash.
```
the same loopback IP get's added to both the externalGW containers? and then we ping the same podIPs inside the cluster?

so instead of doing IPA->podIP and IPB->podIP we do IPLoopBack->podIP from gatewayA and same IPLoopBack->podIP from gatewayB so that the 3 tuple hash for ICMP is the same for dp_hash?

the `-I targetIPs` confuses me because -I is the srcIP here for ingress traffic (-I interface address
Set source address to specified interface address. Argument may be numeric IP address or name of device. When pinging IPv6 link-local address this option is required.) but the variable name is targetIPs which I guess is used in second half of the test

Reply:
ack makes sense thanks tim